components/eventing-controller/controllers/subscription/nats/reconciler_internal_test.go

```diff
@@ -307,23 +310,142 @@ func Test_syncInitialStatus(t *testing.T) {
+			name:              "Return error and default Result{} when backend sync returns error",
+			givenSubscription: testSub,
+			givenReconcilerSetup: func() *Reconciler {
+				te := setupTestEnvironment(t)
+				fakeClient := te.clientBuilder.WithObjects(testSub).Build()
+				te.backend.On("Initialize", mock.Anything).Return(nil)
+				te.backend.On("SyncSubscription", mock.Anything).Return(backendSyncErr)
+				return NewReconciler(ctx, fakeClient, te.backend, happyCleaner, te.logger, te.recorder, defaultSubConfig, happyValidator)
+			},
+			wantReconcileResult: ctrl.Result{},
+			wantReconcileError:  backendSyncErr,
+		},
+		{
+			name:              "Return error and default Result{} when backend delete returns error",
+			givenSubscription: testSubUnderDeletion,
+			givenReconcilerSetup: func() *Reconciler {
+				te := setupTestEnvironment(t)
+				fakeClient := te.clientBuilder.WithObjects(testSubUnderDeletion).Build()
+				te.backend.On("Initialize", mock.Anything).Return(nil)
+				te.backend.On("DeleteSubscription", mock.Anything).Return(backendDeleteErr)
+				return NewReconciler(ctx, fakeClient, te.backend, happyCleaner, te.logger, te.recorder, defaultSubConfig, happyValidator)
+			},
+			wantReconcileResult: ctrl.Result{},
+			wantReconcileError:  backendDeleteErr,
+		},
+		{
+			name:              "Return error and default Result{} when validator returns error",
+			givenSubscription: testSub,
+			givenReconcilerSetup: func() *Reconciler {
+				te := setupTestEnvironment(t)
+				fakeClient := te.clientBuilder.WithObjects(testSub).Build()
+				te.backend.On("Initialize", mock.Anything).Return(nil)
+				te.backend.On("SyncSubscription", mock.Anything).Return(nil)
+				return NewReconciler(ctx, fakeClient, te.backend, happyCleaner, te.logger, te.recorder, defaultSubConfig, unhappyValidator)
+			},
+			wantReconcileResult: ctrl.Result{},
+			wantReconcileError:  validatorErr,
+		},
+		{
+			name:              "Return error and default Result{} when event type cleaner returns error",
+			givenSubscription: testSub,
+			givenReconcilerSetup: func() *Reconciler {
+				te := setupTestEnvironment(t)
+				fakeClient := te.clientBuilder.WithObjects(testSub).Build()
+				te.backend.On("Initialize", mock.Anything).Return(nil)
+				te.backend.On("SyncSubscription", mock.Anything).Return(nil)
+				return NewReconciler(ctx, fakeClient, te.backend, unhappyCleaner, te.logger, te.recorder, defaultSubConfig, happyValidator)
+			},
+			wantReconcileResult: ctrl.Result{},
+			wantReconcileError:  cleanerErr,
+		},
+	}
+
+	for _, testCase := range testCases {
+		tc := testCase
+		t.Run(tc.name, func(t *testing.T) {
+			reconciler := testCase.givenReconcilerSetup()
+			r := ctrl.Request{NamespacedName: types.NamespacedName{
+				Namespace: tc.givenSubscription.Namespace,
+				Name:      tc.givenSubscription.Name,
+			}}
+			res, err := reconciler.Reconcile(context.Background(), r)
+			req.Equal(res, tc.wantReconcileResult)
+			req.Equal(err, tc.wantReconcileError)
```

Comment:
Since `req` is configured with the `t` value from the parent test, I think we would not see the diff of what was expected vs what was computed for each test-case run (in case of errors).

Reply:
@marcobebway I tried it out and the output seems to be ok. It looks like this (I changed the expected error to nil):
```
--- FAIL: TestReconciler_Reconcile (0.00s)
    --- PASS: TestReconciler_Reconcile/Return_nil_and_default_Result{}_when_there_is_no_error_from_the_reconciler_dependencies (0.00s)
    --- PASS: TestReconciler_Reconcile/Return_error_and_default_Result{}_when_backend_sync_returns_error (0.00s)
    --- PASS: TestReconciler_Reconcile/Return_error_and_default_Result{}_when_backend_delete_returns_error (0.00s)
    --- FAIL: TestReconciler_Reconcile/Return_error_and_default_Result{}_when_validator_returns_error (0.00s)



Expected :*errors.errorString(&errors.errorString{s:"invalid sink"})
Actual   :<nil>(<nil>)
```

Reply:
The difference seems to be that if you share the `t` then any failing sub-test would stop the parent test while if you use separate `t`s then after the failure above, the rest of the case would be run as well. I think, here, that difference wouldn't matter anyway.